src/bsoncxx/types/value.cpp

```diff
@@ -96,11 +102,19 @@ bsoncxx::type value::type() const {
 #undef BSONCXX_ENUM
 
 bool operator==(const value& lhs, const value& rhs) {
-    if (lhs.type() != rhs.type()) {
+    if (!lhs.type() && !rhs.type()) {
+        return true;
```

Comment:
So, what this says is that two values with unspecified types compare as equal. Just to make sure we have considered the options, is there any benefit to saying that they don't compare equal (much like NaN != NaN).